cpp/src/binaryop/binaryop.cpp

```diff
@@ -0,0 +1,343 @@
+           cudf::jit::get_type_name(rhs.type()),
+           get_operator_name(op, OperatorType::Direct)})
+      .launch(out.size(), cudf::jit::get_data_ptr(out),
+              cudf::jit::get_data_ptr(lhs), cudf::jit::get_data_ptr(rhs));
+}
+
+void binary_operation(mutable_column_view& out,
+                      column_view const& lhs,
+                      column_view const& rhs,
+                      const std::string& ptx,
+                      cudaStream_t stream) {
+  std::string const output_type_name = cudf::jit::get_type_name(out.type());
+
+  std::string ptx_hash =
+      hash + "." +
+      std::to_string(std::hash<std::string>{}(ptx + output_type_name));
+  std::string cuda_source = "\n#include <cudf/types.hpp>\n" +
+                            cudf::jit::parse_single_function_ptx(
+                                ptx, "GENERIC_BINARY_OP", output_type_name) +
+                            code::kernel;
+
+  cudf::jit::launcher(ptx_hash, cuda_source, header_names, compiler_flags,
+                      headers_code, stream)
+      .set_kernel_inst("cudf::experimental::kernel_v_v",  // name of the kernel
+                                                          // we are launching
+                       {output_type_name,  // list of template arguments
+                        cudf::jit::get_type_name(lhs.type()),
+                        cudf::jit::get_type_name(rhs.type()),
+                        get_operator_name(binary_operator::GENERIC_BINARY,
+                                          OperatorType::Direct)})
+      .launch(out.size(), cudf::jit::get_data_ptr(out),
+              cudf::jit::get_data_ptr(lhs), cudf::jit::get_data_ptr(rhs));
+}
+
+}  // namespace jit
+}  // namespace binops
+
+namespace {
+/**
+ * @brief Computes output valid mask for op between a column and a scalar
+ */
+auto scalar_col_valid_mask_and(column_view const& col,
+                               scalar const& s,
+                               cudaStream_t stream,
+                               rmm::mr::device_memory_resource* mr) {
+  if (col.size() == 0) {
+    return rmm::device_buffer{};
+  }
+
+  if (not s.is_valid()) {
+    return create_null_mask(col.size(), mask_state::ALL_NULL, stream, mr);
+  } else if (s.is_valid() && col.nullable()) {
+    return copy_bitmask(col, stream, mr);
+  } else if (s.is_valid() && not col.nullable()) {
+    return rmm::device_buffer{};
+  }
+}
+}  // namespace
+
+namespace detail {
+
+std::unique_ptr<column> binary_operation(scalar const& lhs,
+                                         column_view const& rhs,
+                                         binary_operator op,
+                                         data_type output_type,
+                                         rmm::mr::device_memory_resource* mr,
+                                         cudaStream_t stream) {
+  // Check for datatype
+  CUDF_EXPECTS(is_numeric(lhs.type()), "Invalid/Unsupported lhs datatype");
```

Comment:
What's the plan for supporting logical binops on non-numeric columns like strings?

Reply:
Which operators do you expect to work? I assume you mean comparison ops (<, >, == etc.), not logic ops (AND/OR etc.)?

Reply:
Yes, sorry, I meant comparison ops.  String column comparisons are already being performed on the device in the unit tests, so I was hoping it would be straightforward to support strings column/scalar comparisons in the binary operation APIs.

Reply:
@jlowe discussed this with @harrism on slack, I'll add string comparison ops to binops in a follow-up PR so we can get this one merged quicker.